<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Exercise-9-10.1" data-toc-modified-id="Exercise-9-10.1-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Exercise 9-10.1</a></span><ul class="toc-item"><li><span><a href="#Question-1" data-toc-modified-id="Question-1-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Question 1</a></span></li><li><span><a href="#Question-2" data-toc-modified-id="Question-2-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Question 2</a></span></li></ul></li><li><span><a href="#Exercise-9-10.2" data-toc-modified-id="Exercise-9-10.2-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Exercise 9-10.2</a></span><ul class="toc-item"><li><span><a href="#Question-1" data-toc-modified-id="Question-1-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Question 1</a></span></li><li><span><a href="#Question-2" data-toc-modified-id="Question-2-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Question 2</a></span></li><li><span><a href="#Question-3" data-toc-modified-id="Question-3-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Question 3</a></span></li><li><span><a href="#Question-4" data-toc-modified-id="Question-4-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Question 4</a></span></li></ul></li><li><span><a href="#Exercise-9-10.3" data-toc-modified-id="Exercise-9-10.3-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Exercise 9-10.3</a></span><ul class="toc-item"><li><span><a href="#Question-1" data-toc-modified-id="Question-1-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Question 1</a></span></li><li><span><a href="#Question-2" data-toc-modified-id="Question-2-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Question 2</a></span></li></ul></li></ul></div>

In [1]:
import numpy as np
import sympy as sp
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats, optimize
import seaborn as sns

config = {
    'figure.facecolor':'w',
    'axes.grid':True,
    'axes.spines.top':False,
    'axes.spines.bottom':False,
    'axes.spines.left':False,
    'axes.spines.right':False,
    'grid.linewidth':0.5,
    'grid.linestyle':'--',
    'figure.constrained_layout.use':True
}

plt.rcParams.update(config)

# Exercise 9-10.1

In 1985, the yield associated to US short term Treasury Notes was about 6%. Assuming that he market expected return is 30% for a portfolio with a beta of 2.

## Question 1

According to the CAPM what is the expected return for the market portfolio?

The CAPM says:

$$\mathbb E[r_i] = r_f + \beta \mathbb E[r_m]$$

We're told that "a portfolo" has an expected return of 30%, and it has a $\beta$ of 2. We're asked to do some algebra and solve for $\mathbb E[r_m]$:

In [7]:
r_f = 0.06
r_i = 0.3
β = 2
print(f'CAPM estimated market return: {(r_i - r_f) / β:0.2%}')

CAPM estimated market return: 12.00%


## Question 2

The stock Astrotech has been sold 40\\$. An analyst forecast a dividend of 3\\$ in 1988 and an ask price of 41\\$. If the forecast of the analyst are true and β = −0.4, was the stock over or under priced?

First, compute the expected price under the CAPM:

In [8]:
r_i = r_f + -0.4 * 0.12
r_i

0.011999999999999997

This will be the expected increase in the stock price to next year. To tie this into the price, we can think about the return as the percent change in the stock price, including the dividend. That is:

$$r_i = \frac{P_1 + D_1}{P_0} - 1$$

Note that this is just percent change from $t=0$ to $t=1$, adjusting for the dividend payout. We need to solve for $P_0$:

$$P_0 = \frac{P_1 + D_1}{1 + r_i}$$

In [15]:
print(f'Implied price at P0: {(41 + 3) / (1 + r_i):0.2f}$')

Implied price at P0: 43.48$


Since the implied price of 43.48 is greater than the actual asking price of 41, we conclude that the stock was under-priced.

# Exercise 9-10.2

The risk-free asset return is 2.5%. You can invest in a private equity firm whose return and
associated probability are presented in the following table:

|              |   1  |   2  |
|--------------|------|------|
| Return       | +25% | -50% |
| Probability  | 70%  | 30%  |

## Question 1

Compute the variance and the expected return of the investment

In [21]:
r_f = 0.025
p = np.array([0.7, 0.3])
r = np.array([0.25, -0.5])
μ = (r * p).sum()
σ2 = (r ** 2 * p).sum() - μ ** 2
print(f'Expected return: {μ:0.3f}')
print(f'Variance: {σ2:0.3f}')
print(f'Volatility: {np.sqrt(σ2):0.3f}')

Expected return: 0.025
Variance: 0.118
Volatility: 0.344


## Question 2

Assuming that your behaviour can be modelled by a mean-variance utility function, and that you do not have any extra information about this investment, which part of you wealth could you invest in?

We're goning to make a portfolio between this asset and the risk-free asset. Call the weight on the risky asset $w$, then we will have:

$$\begin{align}\mu_p &= w\mu + (1 - w) r_f \\
    \sigma_p^2 &= w^2 \sigma^2 \end{align}$$
    
So utility is:

$$\begin{align}U &= \mu_p - \frac{A}{2}\sigma_p^2 \\
                 &= w\mu_p + (1 - w)r_f - \frac{A}{2}w^2 \sigma^2 \end{align}$$
                 
Take derivative and solve:

$$w = \frac{\mu - r_f}{A\sigma^2_p}$$

We don't know our $A$, but it doesn't matter! Notice that $r_f = 2.5\%$, and we computed $\mu_p = 2.5\%$. So no matter what the risk tolerance, we choose $w=0$ -- we don't want this portfolio at all!

## Question 3

Now we are told that:

- Expected Market Return $E[R_m]$ = 6.5%
- Market Variance $V[R_m]$ = 0.0289
- Covariance between the market and the asset $\sigma_{iM}$ = -0.00289

From this information, we can compute $\beta = \frac{\sigma_{i,M}}{\sigma^2_M}$:

In [23]:
σ_im = -0.00289
σ2_m = 0.0289
μ_m = 0.065

β = σ_im / σ2_m
print(f'CAPM beta for this asset: {β:0.3f}')

CAPM beta for this asset: -0.100


And we can compute the expected returns of this asset, given this beta with the market:
$$\mathbb E [r_i] = r_f + \beta (\mathbb E[r_m] - r_f)$$

In [26]:
Er_i = r_f + β * (μ_m - r_f)
Er_i

0.021

Under the CAPM, we compute that the expected return is 2.1%. Notice that this is lower than the observed return of 2.5%. This suggests that the asset is underpriced, so it is rational to buy it. This is despite the fact that, in any case, the return is lower than the risk-free rate. 

## Question 4

Determine the threshold $\sigma_{i,M}$ from which you will stop investing in this private equity firm.

We will stop if $\mathbb E[r_i] = r_i = 0.025$, so we just solve for $\sigma_{i,M}$ such that:

$$0.025 = r_f + \frac{\sigma_{i,M}}{\sigma^2_M}(\mu_m - r_f)$$

In [30]:
σ_im_stop = (0.025 - r_f) * σ2_m / (μ_m - r_f)
print(f'σ_im at which we stop investing: {σ_im_stop:0.3f}')

σ_im at which we stop investing: 0.000


How should be interpret this? Remember that this is the **covariance**, not the variance, so we shouldn't be shocked that it's zero. Remember that the beta with the market was negative. Basically, we like this asset because it provides a downside hedge against market declines -- when the market goes down, this asset goes up. If it stops serving that function -- if the beta becomes positive -- we stop being interested in this asset.

# Exercise 9-10.3

The asset price at date 0 is denoted $P_0$. The asset bid price at date 1 is a random variable denoted $P_1$. Assuming the CAPM framework is valid and that the beta of the asset is denoted $\beta$.

## Question 1

Prove that $P_0 = \frac{E(\tilde{P}_1)}{(1 + r_f + (\mu_M - r_f)\beta)}$

First, we note that $\mathbb E[P_1] = (1 + \mathbb E[r_1])P_0$. Solving for $P_0$, we get:

$$P_0 = \frac{\mathbb E[P_1]}{(1 + \mathbb E[r_1])}$$

Great, we're already most of the way there. Now we just need to know what is $\mathbb E[r_1]$. We're told that the CAPM framework is valid, and that tells us that:

$$\mathbb E[r_1] = r_f + \beta (\mathbb E[r_m] - r_f)$$

Define $\mu_m = \mathbb E[r_m]$, then just plug this expression into our fraction:

$$P_0 = \frac{\mathbb E[P_1]}{(1 + r_f + \beta (\mu_m - r_f))}$$

The expression is verified.

## Question 2

Show that we can re-write the above expression to:

$$P_0 = \frac{\mathbb E[P_1] - a \rho \sigma}{1 + r_f}$$

Where $\rho$ is the correlation between the asset and the market, $\sigma$ is the voltility of the asset, and $a$ is an unknown parameter.

Let's start from the CAPM expression:

$$\mathbb E[r_1] = r_f + \beta (\mathbb E[r_m] - r_f)$$

We know that we can write $\beta = \frac{\sigma_{i,m}}{\sigma^2_m}$, because we can think about the CAPM as a linear regression. Furthermore, we know that $\sigma_{i,m} = \rho \sigma_i \sigma_m$. So, after cancellation, our expression becomes:

$$\mathbb E[r_1] = r_f + \frac{\rho \sigma_i}{\sigma_m} (\mathbb \mu_m - r_f)$$

Plug this into the percent change equation:

$$P_0 = \frac{\mathbb E[P_1]}{1 + r_f + \frac{\rho \sigma_i}{\sigma_m} (\mu_m - r_f)}$$

Define $S_m = \frac{\mu_m - r_f}{\sigma_m}$ as the Sharpe ratio of the market. This gives:

$$P_0 = \frac{\mathbb E[P_1]}{1 + r_f + S_m \rho \sigma_i}$$

Now shuffle some stuff around:

$$\begin{align}
P_0(1 + r_f + S_m \rho \sigma_i) &= \mathbb E[P_1] \\
(1 + r_f) P_0 + S_m \rho \sigma_i P_0 &= \mathbb E[P_1] \\
\end{align}$$

We really want to get rid of the $P_0$ in the 2nd summand on the left-hand side. If we could, we would be done. The final twist in the derivation will be to look inside $sigma_i$ for some help. We know that:

$$\begin{align}
\sigma_i &= \sqrt{\text{Var}(r_i)} \\ 
&= \sqrt{\text{Var}\left(\frac{P_1}{P_0} - 1 \right)} \\
&= \sqrt{\text{Var}\left(\frac{P_1}{P_0} - \frac{P_0}{P_0} \right)} 
\end{align}$$

Standing here at time $t=0$, $P_0$ is non-random, so we can factor it out. When we do, we have to remember that variance is a quadratic function, so what comes out will be $P_0^2$, not $P_0$:

$$\begin{align}
\sigma_i &= \sqrt{\frac{1}{P_0^2}\text{Var}\left(P_1 - P_0 \right)}  \\
&= \frac{1}{P_0}\sqrt{\text{Var}\left(P_1 - P_0 \right)} \\
\sigma_i &= \frac{1}{P_0}\tilde{\sigma}_i
\end{align}$$

Plug this into our expression:

$$(1 + r_f) P_0 + S_m \rho \frac{P_0 \tilde{\sigma}_i}{P_0} = \mathbb E[P_1]$$

So after rearranging:

$$P_0 = \frac{\mathbb E[P_1] - S_m \rho \tilde{\sigma}_i}{1 + r_f}$$

The last bit of cleanup is to think more closely about $\tilde \sigma_i = \sqrt{\text{Var}\left(P_1 - P_0 \right)}$. Write this out using the definition of variance:

$$\begin{align}
\tilde \sigma_i^2 &= \mathbb E[((P_1 - P_0) - \mathbb E[(P_1 - P_0)])^2] \\
                  &= \mathbb E[((P_1 - \mathbb E[P_1] - P_0 + \mathbb E[P_0]))^2] \\
                  &= \mathbb E[(P_1 - \mathbb E[P_1])^2] \\
                  &= \text{Var}(P_1)
\end{align}$$

So we find that $\tilde \sigma_i$ is the volatility in the **price** of the asset, rather than in the **returns** of the asset.